pkg/sync/file/filepath_sync.go

```diff
@@ -0,0 +1,140 @@
+package file
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/open-feature/flagd/pkg/sync"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/open-feature/flagd/pkg/logger"
+)
+
+type Sync struct {
+	URI          string
+	Logger       *logger.Logger
+	ProviderArgs sync.ProviderArgs
+	// FileType indicates the file type e.g., json, yaml/yml etc.,
+	fileType string
+}
+
+//nolint:funlen
+func (fs *Sync) Sync(ctx context.Context, dataSync chan<- sync.DataSync) error {
+	fs.Logger.Info("Starting filepath sync notifier")
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		fs.Logger.Fatal(err.Error())
+	}
+	defer watcher.Close()
+
+	err = watcher.Add(fs.URI)
+	if err != nil {
+		fs.Logger.Error(err.Error())
+		return err
+	}
+
+	// file watcher is ready(and stable), fetch and emit the initial results
+	fetch, err := fs.fetch(ctx)
+	if err != nil {
+		fs.Logger.Error(err.Error())
+		return err
+	}
+
+	dataSync <- sync.DataSync{FlagData: fetch, Source: fs.URI}
+
+	fs.Logger.Info(fmt.Sprintf("Watching filepath: %s", fs.URI))
+	for {
+		select {
+		case event, ok := <-watcher.Events:
+			if !ok {
+				fs.Logger.Info("Filepath notifier closed")
+				return errors.New("filepath notifier closed")
+			}
+
+			switch event.Op {
+			case fsnotify.Create:
+				fs.Logger.Debug("New configuration created")
+				msg, err := fs.fetch(ctx)
+				if err != nil {
+					fs.Logger.Error(err.Error())
+				}
+				dataSync <- sync.DataSync{
+					FlagData: msg,
+					Source:   fs.URI,
+				}
+			case fsnotify.Write:
+				fs.Logger.Debug("Configuration modified")
+				msg, err := fs.fetch(ctx)
+				if err != nil {
+					fs.Logger.Error(err.Error())
+				}
+				dataSync <- sync.DataSync{
+					FlagData: msg,
+					Source:   fs.URI,
+				}
+			case fsnotify.Remove:
+				// K8s exposes config maps as symlinks.
+				// Updates cause a remove event, we need to re-add the watcher in this case.
+				err = watcher.Add(fs.URI)
+				if err != nil {
+					fs.Logger.Error(fmt.Sprintf("Error restoring watcher, file may have been deleted: %s", err.Error()))
+				}
+			}
+			fs.Logger.Info(fmt.Sprintf("Filepath event: %s %s", event.Name, event.Op.String()))
+		case err, ok := <-watcher.Errors:
+			if !ok {
+				return errors.New("watcher error")
+			}
+			fs.Logger.Error(err.Error())
+		case <-ctx.Done():
+			fs.Logger.Debug("exiting file watcher")
```

Comment:
We should aim for consistency on whether capitalizing or not logs

Reply:
Yes, I will revisit logs and try to unify them